Project/Project/Controllers/UserController.cs

```diff
@@ -79,12 +90,82 @@ public async Task<IActionResult> Update(AccountViewModel model)
             _context.Users.Update(user);
             await _context.SaveChangesAsync();
 
+            var userWithoutPassword = new User
+            {
+                Id = user.Id,
+                Name = user.Name,
+                Email = user.Email,
+            };
+
+            HttpContext.Session.SetString("CurrentUser", JsonConvert.SerializeObject(userWithoutPassword));
+
             TempData["SuccessMessage"] = "Account updated successfully";
 
             return RedirectToAction("Update", "User");
 
         }
         return RedirectToAction("Login", "Auth");
     }
+
+    [HttpPost]
+    [HttpPost]
+    public async Task<IActionResult> Delete()
+    {
+        var userJson = HttpContext.Session.GetString("CurrentUser");
+
+        if (userJson == null)
+        {
+            var originalUrl = Request.Headers["Referer"].ToString();
+            return RedirectToAction("Login", "Auth", new { returnUrl = originalUrl });
+        }
+
+        var currentUser = JsonConvert.DeserializeObject<User>(userJson);
+        if (currentUser == null)
+        {
+            return View("NotFound");
+        }
+
+        // Start a transaction
+        using (var transaction = await _context.Database.BeginTransactionAsync())
+        {
+            try
+            {
+                var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == currentUser.Id);
+
+                if (user == null)
+                {
+                    return View("NotFound");
+                }
+
+                // Remove user-related data
+                _context.Users.Remove(user);
+
+                // Commit transaction
+                await _context.SaveChangesAsync();
+                await transaction.CommitAsync();
+
+                // Clear session
+                HttpContext.Session.Clear();
+
+                // Log the successful deletion
+                _logger.LogInformation($"User {currentUser.Id} deleted successfully.");
+
+                TempData["SuccessMessage"] = "Account successfully deleted!";
+                return RedirectToAction("Index", "Home");
+            }
+            catch (Exception ex)
+            {
+                // Rollback transaction in case of error
+                await transaction.RollbackAsync();
+
+                // Log the error
+                _logger.LogError(ex, $"Error deleting user {currentUser.Id}.");
+
+                // Return an error view or message
+                TempData["ErrorMessage"] = "An error occurred while deleting the account.";
```

Comment:
An error occurred while deleting the account. Please try again later or reach out to us via [Link to Impress page]